src/vs/editor/contrib/gotoSymbol/browser/peek/referencesController.ts

```diff
@@ -227,7 +232,7 @@ export abstract class ReferencesController implements IEditorContribution {
 	}
 
 	closeWidget(focusEditor = true): void {
-		this._widget?.dispose();
+		this._widget?.disposeFromClose();
```

Comment:
Not a fan of having `disposeFromClose` and `dispose`. Could we instead change this method to only run once, like have a boolean `_isClosing` and exit early if this is set because closing once is enough.

Reply:
Fixed!